fdbcli/DebugCommands.actor.cpp

```diff
@@ -320,28 +322,45 @@ ACTOR Future<bool> checkallCommandActor(Database cx, std::vector<StringRef> toke
 				for (const auto& server : servers) {
 					printf("  %s\n", server.address().toString().c_str());
 				}
-				wait(store(version, getVersion(cx)));
 				state std::vector<Future<ErrorOr<GetKeyValuesReply>>> replies;
-				for (const auto& s : keyServers[i].second) { // for each storage server
-					GetKeyValuesRequest req;
-					req.begin = begin;
-					req.end = end;
-					req.limit = CLIENT_KNOBS->KRM_GET_RANGE_LIMIT;
-					req.limitBytes = CLIENT_KNOBS->KRM_GET_RANGE_LIMIT_BYTES;
-					req.version = version;
-					req.tags = TagSet();
+				state bool hasMore = true;
+				state int round = 0;
+				while (hasMore) {
+					wait(store(version, getVersion(cx)));
+					printf("Start a new round %d\n", round);
+					replies.clear();
+					for (const auto& s : keyServers[i].second) { // for each storage server
+						// printf("For each SS %d, range.begin is %s\n", s.uniqueID, printable(range).c_str());
```

Comment:
if I uncomment this line, there would be a immediate seg fault as below. 
```
Start a new round 0
For each SS -930692084, range.begin is SIGNAL: Segmentation fault (11)
```


Otherwise, it can go for a round or two before seg fault, 
```
Start a new round 0
For each SS -930692084
Sent request for SS -930692084
For each SS -1357485812
Sent request for SS -1357485812
For each SS -1701186485
Sent request for SS -1701186485
For each SS 2010161760
Sent request for SS 2010161760
For each SS -660526461
Sent request for SS -660526461
For each SS 1194619847
Sent request for SS 1194619847
waiting for 6 replies at version: 59247269453790
Done waiting in round 0
Warning: Consistency check was incomplete, last key of server 0 that was checked: \x158\x19\x02\xb9\x8a\xf4B\x15\xdc\x15>\x15\x01\x15\x08\x13\xfe
AllSame 1, hasMore 1, checkAll 0
Start a new round 1
For each SS -930692084
Sent request for SS -930692084
For each SS -1357485812
Sent request for SS -1357485812
For each SS -1701186485
Sent request for SS -1701186485
For each SS 2010161760
Sent request for SS 2010161760
For each SS -660526461
Sent request for SS -660526461
For each SS 1194619847
Sent request for SS 1194619847
waiting for 6 replies at version: 59247271927521

WARNING: Long delay (Ctrl-C to interrupt)
Done waiting in round 1
Warning: Consistency check was incomplete, last key of server 0 that was checked: \x158\x19\x02\xb9\x8b\x09Q\x15"\x00\x15\x02\x15\x0e\x15\x10\x022F114CBC-E663-4DE9-A193-D1724896E516\x00\x02MkYxMTRDQkMtRTY2My00REU5LUExOTMtRDE3MjQ4OTZFNTE2\x00
AllSame 1, hasMore 1, checkAll 0
Start a new round 2
For each SS -930692084
Sent request for SS -930692084
```

@jzhou77 do you see any obvious reason that why `printable(range).c_str()` cannot be printed, or why there is seg fault at all?

Reply:
i tested it with command `checkall \x158\x19\x02\xb9\x8a\xe4s\x15s\x00\x15\x02\x15\x08\x17Z\xc3Z\x17\x11\xea \x158\x19\x02\xb9\x8c\x1e\x18\x15/\x00\x15\x01\x15=\x0258`

Reply:
this is because `KeySelectorRef` has a `KeyRef`, whose memory is in `replies` and is cleared. will fix it soon.